h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -70,7 +70,10 @@
   @Override public boolean haveMojo() { return true; }
 
   @Override protected void checkMemoryFootPrint() {
-    long mem_usage = 8 /*doubles*/ * (_parms._k * _train.numCols() + _parms._k );  // loose estimation of memory usage
+    HeartBeat hb = H2O.SELF._heartbeat;
+    double p = hex.util.LinearAlgebraUtils.numColsExp(_train,true);
+    long mem_usage = (long)(hb._cpus_allowed * p*_parms._k * 8 * Math.log((double)_train.lastVec().nChunks())/
```

Comment:
Tomas:  I looked at the other checkMemoryFootPrint() in deep learning and elsewhere, they all use the H2O.SELF._heartbeat call and then you can call get_free_mem() or other functions to get what you want.  Please advice on proper use because I have no idea.